dev/snippets/README.md

```diff
@@ -1,59 +1,136 @@
+
+The code `snippet` tool can expand sample code into full Flutter applications.
+These sample applications can be directly copied and used to demonstrate the
+API's functionality in a sample application:
+
+```dart
+/// {@tool snippet --template=stateless_widget_material}
+/// This example shows how to make a simple [FloatingActionButton] in a
+/// [Scaffold], with a pink [backgroundColor] and a thumbs up [Icon].
 ///
-///   @override
-///   Widget build(BuildContext context) {
-///     return Container(color: const Color(0xFF2DBD3A));
-///   }
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     appBar: AppBar(
+///       title: Text('Floating Action Button Sample'),
+///     ),
+///     body: Center(
+///       child: Text('Press the button below!')
+///     ),
+///     floatingActionButton: FloatingActionButton(
+///       onPressed: () {
+///         // Add your onPressed code here!
+///       },
+///       child: Icon(Icons.thumb_up),
+///       backgroundColor: Colors.pink,
+///     ),
+///   );
 /// }
 /// ```
 /// {@end-tool}
 ```
 
-And converts it into something which has a nice visual presentation, and 
-a button to automatically copy the sample to the clipboard.
+This uses the skeleton for [application](config/skeletons/application.html)
+snippets.
 
-It does this by processing the source input and emitting HTML for output,
-which dartdoc places back into the documentation. Any options given to the
- `{@tool ...}` directive are passed on verbatim to the tool.
+Code `snippets` also allow for quick Flutter app generation using the following command:
 
-To render the above, the snippets tool needs to render the code in a combination
-of markdown and HTML, using the `{@inject-html}` dartdoc directive.
+```bash
+flutter create --sample=[directory.File.sampleNumber] [name_of_project_directory]
+```
 
-## Templates
+#### Templates
 
 In order to support showing an entire app when you click on the right tab of
-the code snippet UI, we have to be able to insert the snippet into the template
+the code snippet UI, we have to be able to insert the `snippet` into the template
 and instantiate the right parts.
 
 To do this, there is a [config/templates](config/templates) directory that
 contains a list of templates. These templates represent an entire app that the
-snippet can be placed into, basically a replacement for `lib/main.dart` in a
+`snippet` can be placed into, basically a replacement for `lib/main.dart` in a
 flutter app package.
 
+For more information about how to create, use, or update templates, see
+[config/templates/README.md](config/templates/README.md).
+
 ## Skeletons
 
+The code block generation tools process the source input and emit HTML for output,
+which dartdoc places back into the documentation. Any options given to the
+ `{@tool ...}` directive are passed on verbatim to the tool.
+
+To render the these examples, the `snippets` tool needs to render the code in a
```

Comment:
'the these'

Reply:
Fixed it! Also re-ordered the skeleton implementation details to come after the definition of a skeleton, since that makes more sense